File: Recursion/Reverse String/code.cpp
#include<bits/stdc++.h>
using namespace std;

void reverse(string& str, int i, int j ) {
	if (i > j)
		return ;

	swap(str[i], str[j]);

	reverse(str, i+1, j-1);
}

int main() {
	string str = "Hello World";
	reverse(str, 0, str.length()-1);
	cout << str << endl;
	return 0;
}

File: Recursion/Reverse String/code.py
def reverseStr(string):
	return getReversed(string, 0, len(string)-1)

def getReversed(string, l, r):
	if l > r:
		return

	print(string[l], string[r])
	# tmp = string[l]
	# string[l] = string[r]
	# string[r] = tmp
	
	getReversed(string, l+1, r-1)
	return string

print(reverseStr("Hello World"))